draft-tschofenig-rats-psa-token.md

```diff
@@ -365,7 +368,7 @@ The EAT `ueid` (claim key 256) of type RAND is used.  The following constraints
 apply to the `ueid-type`:
 
 * The length MUST be 33 bytes.
-* The first byte MUST be 0x01 (RAND) followed by the 32-bytes key hash.
+* The first byte MUST be 0x01 (RAND) followed by 32 random bytes.  {{PSA-API}} provides implementation options for obtaining the random bytes from the underlying IAK.
```

Comment:
Why is this new sentence useful? Seems like a distraction

Reply:
I first thought that the change made sense: those are random bytes, `{{PSA-API}}` just happens to derive those bytes from the IAK.

But looking at the context it's actually the other way around, it's the IAK unique identifier so the thing that makes most sense is that it is derived from it. Those are not random bytes, it just happens that the key is generated randomly and we reuse the  `RAND` option of the `ueid` claim.

What do you think about:

```
* The first byte MUST be 0x01 (RAND) followed by a 32-bytes value of the IAK unique identifier ({{PSA-API}} provides several implementation options for deriving the unique identifier from the IAK).
```

Reply:
The thing here is we are using EAT's [`ueid` of type `RAND`](https://www.ietf.org/archive/id/draft-ietf-rats-eat-24.html#table-1), which requires the bytes to be _"[a] 128, 192 or 256-bit random number generated once and stored in the entity."_

Reply:
The next sentence is `This may be constructed by concatenating enough identifiers to make up an equivalent number of random bits and then feeding the concatenation through a cryptographic hash function.`. So it seems already inconsistent to say that those must be random bytes (and incidentally that is what we are doing in PSA).

My guess is that the intent in comparison to the other `ueid` options is to say the bytes have no meaning (they look random) but they contain enough entropy to have some assurances of uniqueness.

Semantically though, this specification mandates that the Instance ID should be the IAK unique identifier, not a distinct unique random value. I feel the current wording makes it sound like the `{{PSA-API}}` implementation is optional and using a different random value is acceptable.

Reply:
> I feel the current wording makes it sound like the {{PSA-API}} implementation is optional and using a different random value is acceptable.

OK, got it.

Slightly rewording your proposal above:

_"The first byte MUST be 0x01 (RAND) followed by a 32-byte value of the IAK unique identifier.  {{PSA-API}} provides implementation options for deriving the unique identifier from the IAK."_

Reply:
Your rewording is better than my proposal.

Reply:
> Your rewording is better than my proposal.

I have slightly reworded my rewording at https://github.com/thomas-fossati/draft-psa-token/pull/129/files#r1442286052